File: app/src/main/java/in/jackz/kotlinify/SecondActivity.java
package in.jackz.kotlinify;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        Intent i=getIntent();
        User user= (User) i.getSerializableExtra("data");
        Log.d("hai",user.getId());


    }
}

File: app/src/main/java/in/jackz/kotlinify/SecondActivity.kt
package `in`.jackz.kotlinify

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.View
import android.widget.Button

class SecondActivity : AppCompatActivity() {

    internal var mButton: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        val i =intent
        val (id) = i.getSerializableExtra("data") as User
        Log.d("hai", id)
        mButton!!.setOnClickListener(onClickListener)

    }

    internal var onClickListener: View.OnClickListener = View.OnClickListener { }
}
